archive_upload/handlers/dsmc_handlers.py

```diff
@@ -0,0 +1,754 @@
+        Job is run in the background to be polled by the status endpoint.
+
+        :param runfolder_archive: the archive we want to re-upload
+        :return: HTTP 200 if nothing to reupload, HTTP 202 if reupload started successfully, with a `job_id` to be used for later polling,
+                 HTTP 500 if unexpected error detected.
+
+        """
+        monitored_dir = self.config["path_to_archive_root"]
+        helper = ReuploadHelper()
+
+        if not self._validate_runfolder_exists(runfolder_archive, monitored_dir):
+            msg = "Error when validating runfolder. {} is not found under {}.".format(runfolder_archive, monitored_dir)
+            self.abort(msg)
+            return
+
+        path_to_archive = os.path.join(monitored_dir, runfolder_archive)
+        dsmc_log_root_dir = self.config["log_directory"]
+
+        if not self._is_valid_log_dir(dsmc_log_root_dir):
+            msg = "Error when validating log dir. {} is not a directory.".format(dsmc_log_root_dir)
+            self.abort(msg)
+            return
+
+        dsmc_log_dir = "{}/dsmc_{}".format(dsmc_log_root_dir, runfolder_archive)
+
+        if not os.path.exists(dsmc_log_dir):
+            os.makedirs(dsmc_log_dir)
+
+        # Fetch the description of the last uploaded version of this archive
+        descr = helper.get_pdc_descr(path_to_archive, dsmc_log_dir)
+
+        # Get the local and remote filelist, and then get the list of files
+        # that are missing on remote side, or differs in byte size.
+        # NB. Uploaded list contains folders as well, but when we check local
+        # content we only look at the files, and ignore the folders.
+        uploaded_files = helper.get_pdc_filelist(path_to_archive, descr, dsmc_log_dir)
+        local_files = helper.get_local_filelist(path_to_archive)
+        reupload_files = helper.get_files_to_reupload(local_files, uploaded_files)
+
+        # Upload the missing files with the same description previously used.
+        if reupload_files:
+            job_id = helper.reupload(reupload_files, descr, dsmc_log_dir, self.runner_service)
+            log.debug("Reupload job_id {}".format(job_id))
+
+            status_end_point = "{0}://{1}{2}".format(
+            self.request.protocol,
+            self.request.host,
+            self.reverse_url("status", job_id))
+
+            response_data = {
+                "job_id": job_id,
+                "service_version": version,
+                "link": status_end_point,
+                "state": State.STARTED,
+                "dsmc_log_dir": dsmc_log_dir}
+
+            self.set_status(202, reason="started reuploading")
+        else:
+            log.debug("Nothing to do - everything already uploaded.")
+
+            response_data = {
+            "service_version": version,
+            "state": State.DONE,
+            "dsmc_log_dir": dsmc_log_dir}
+
+            self.set_status(200, reason="nothing to reupload")
```

Comment:
Is this a really a 200? I'm just thinking that if there is a need to reupload, we expect that some files should be uploaded? I'm unsure here.

Reply:
I'm not really sure either. But your argument makes sense, so lets set to 400 for now.